packages/browser-agent-core/features/spa/aggregate/index.js

```diff
@@ -0,0 +1,708 @@
+        if (!state.currentNode) {
+          interaction.checkFinish()
+          if (state.depth) setCurrentNode(interaction.root)
+        }
+      }, undefined, baseEE)
+
+
+      register(INTERACTION_API + 'actionText', function (t, actionText) {
+        var customAttrs = this.ixn.root.attrs.custom
+        if (actionText) customAttrs.actionText = actionText
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'setName', function (t, name, trigger) {
+        var attrs = this.ixn.root.attrs
+        if (name) attrs.customName = name
+        if (trigger) attrs.trigger = trigger
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'setAttribute', function (t, name, value) {
+        this.ixn.root.attrs.custom[name] = value
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'end', function (timestamp) {
+        var interaction = this.ixn
+        var node = activeNodeFor(interaction)
+        setCurrentNode(null)
+        node.child('customEnd', timestamp).finish(timestamp)
+        interaction.finish()
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'ignore', function (t) {
+        this.ixn.ignored = true
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'save', function (t) {
+        this.ixn.save = true
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'tracer', function (timestamp, name, store) {
+        var interaction = this.ixn
+        var parent = activeNodeFor(interaction)
+        var ctx = baseEE.context(store)
+        if (!name) {
+          ctx.inc = ++interaction[REMAINING]
+
+          return (ctx[SPA_NODE] = parent)
+        }
+        ctx[SPA_NODE] = parent.child('customTracer', timestamp, name)
+      }, undefined, baseEE)
+
+      register(FN_START, tracerDone, undefined, tracerEE)
+      register('no-' + FN_START, tracerDone, undefined, tracerEE)
+
+      function tracerDone(timestamp, interactionContext, hasCb) {
+        var node = this[SPA_NODE]
+        if (!node) return
+        var interaction = node[INTERACTION]
+        var inc = this.inc
+        this.isTraced = true
+        if (inc) {
+          interaction[REMAINING]--
+        } else if (node) {
+          node.finish(timestamp)
+        }
+        hasCb ? setCurrentNode(node) : interaction.checkFinish()
+      }
+
+      register(INTERACTION_API + 'getContext', function (t, cb) {
+        var store = this.ixn.root.attrs.store
+        setTimeout(function () {
+          cb(store)
+        }, 0)
+      }, undefined, baseEE)
+
+      register(INTERACTION_API + 'onEnd', function (t, cb) {
+        this.ixn.handlers.push(cb)
+      }, undefined, baseEE)
+
+      register('api-routeName', function (t, currentRouteName) {
+        state.lastSeenRouteName = currentRouteName
+        if (state.currentNode) state.currentNode[INTERACTION].setNewRoute(currentRouteName)
```

Comment:
This looks like it was added post 1216, was this to address a bug found in testing?